lib/src/rules/prefer_final_locals.dart

```diff
@@ -76,6 +81,101 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   _Visitor(this.rule);
 
+  void checkPatternFields(DartPattern node) {
+    var function = node.thisOrAncestorOfType<FunctionBody>();
+    if (function == null) return;
+
+    late NodeList<PatternField> fields;
+    if (node is RecordPattern) fields = node.fields;
+    if (node is ObjectPattern) fields = node.fields;
+
+    var parent = node.unParenthesized.parent;
```

Comment:
I don't think this is doing what you expect it to do. If the `node` is a `ParenthesizedPattern`, then `unParenthesized` will return the expression inside the parentheses, and `parent` will return the `ParenthesizedPattern`. I think you either want `node.parent` or a loop that will get you to the parent of any enclosing parenthesized patterns.

Reply:
Addressed in https://github.com/dart-lang/linter/pull/4225